modules/Microsoft.Network/virtualNetworkGateways/deploy.bicep

```diff
@@ -251,7 +254,7 @@ var ipConfiguration = isActiveActiveValid ? [
   }
 ]
 
-var vpnClientConfiguration = {
+var vpnClientConfiguration = !empty(clientRootCertData) ? {
```

Comment:
Not sure I fully get the logic as I notice a nested condition on the same parameter at row 263. 

Also, it looks `vpnClientConfiguration` is set differently based on if either `clientRootCertData `or `vpnClientAadConfiguration` are empty. If that is the case for readability I'd probably suggest:
- to mention that in the parameter description (I think the 2 parameters cannot be both set)
- to create 2 different variables based on which value is empty
- to set the module parameter to the corresponding variable accordingly

Reply:
Descriptions added to parameters to keep current functionality

Reply:
I'd still suggest to at least remove the condition on `vpnClientRootCertificates` at row 262, since we're checking on the empty value of `clientRootCertData`, which is redundant since the same condition is checked at row 256

Reply:
Removed the duplicate check in latest commit